### PR TITLE
Improve panos_commit module

### PR DESCRIPTION
**Pull request by traittinen**
##### SUMMARY

Adding new features for panos_commit module
- Add description/comment for commit job
- Run commit only on changes done by specified admin
- Run commit only for specific VSYS
- Return commit job ID and API status information instead of "okey dokey"

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

panos_commit

##### ADDITIONAL INFORMATION

Sample usage:

```yaml
   - name: Commit configuration on FW
      panos_commit:
        ip_address: "{{ pan_host }}"
        username: "{{ pan_user }}"
        password: "{{ pan_pw }}"
        interval: 2.0
        timeout: 90
        description: "Added by Ansible network automation"
        commit_vsys: "vsys{{ vsys_id }}"
```

### CODE REVIEW

File: lib/ansible/modules/network/panos/panos_commit.py [hunk 1]
```diff
@@ -78,25 +140,35 @@
 
 def main():
     argument_spec = dict(
-        ip_address=dict(),
-        password=dict(no_log=True),
-        username=dict(default='admin'),
+        ip_address=dict(required=True, type='str'),
+        password=dict(fallback=(env_fallback, ['ANSIBLE_NET_PASSWORD']), no_log=True),
+        username=dict(fallback=(env_fallback, ['ANSIBLE_NET_USERNAME'])),
```
Comment (shinmog) [inline lib/ansible/modules/network/panos/panos_commit.py]: This change will break existing playbooks if they rely on the default username being `'admin'`.

**Review by shinmog — CHANGES_REQUESTED** [review]
@traittinen 

This is a cool update, thanks :)

However, you are adding a dependency of `lxml` in this implementation that does not already exist for the panos modules.  This needs to be changed to use `xml.etree.ElementTree` like pan-python does.

Also, does committing changes for a specific admin code work for Panorama as well?  I haven't played with partial commits yet, so I don't know the answer.

File: lib/ansible/modules/network/panos/panos_commit.py (hunk 1 above)
Comment (traittinen) [reply lib/ansible/modules/network/panos/panos_commit.py]: I will add "admin" as last fallback option for backward compatibilty.

**Comment by traittinen** [comment]
@shinmog 
> However, you are adding a dependency of `lxml` in this implementation that does not already exist for the panos modules. This needs to be changed to use `xml.etree.ElementTree` like pan-python does.

I understand that it's better to use `xml.etree.ElementTree` in this case since adding a new dependency for existing module could break something when upgrading to new version of Ansible. However, would it be possible to utilize lxml in new modules? I'm asking because I have ready panos_facts module I could possibly share, but that's using lxml (xpath) quite heavily.

> Also, does committing changes for a specific admin code work for Panorama as well? I haven't played with partial commits yet, so I don't know the answer.

Yes, it seems to work.

**Comment by traittinen** [comment]
@shinmog are you happy with the changes?

**Review by shinmog — APPROVED** [review]
lgtm 👍

**Comment by shinmog** [comment]
@traittinen 

To answer your question about lxml in future modules, I think no, unfortunately.  lxml provides some niceties, which is not a high enough bar to warrant adding another dependency to the panos modules.

**Comment by ganeshrn** [comment]
rebuild_merge

**Pull request merged**